### DISCUSSION TITLE
Where to start with image conversion? Webp to png, confused on how to setup anything at all, readme's and stackoverflow have led me nowhere

### DISCUSSION THREAD

**Discussion by pigzit** [ImageMagick/ImageMagick#5918] [Help] [answered]
So I have a folder of webp images, and I just ran the imagemagick installer and downloaded the webp "dwebp decoder" tool zip file from [here](https://developers.google.com/speed/webp/download?hl=en). Though reading the readme in the dwebp zip folder has led me to a dead end.
I'm not sure what to do with my installed imagemagick program, I'm not really sure what I installed, all that can open for me is an image viewer, and I don't know where to go from here.
Reading through the website and the webp documentation hasn't been too helpful either, and most stackoverflow posts like [this one](https://stackoverflow.com/questions/49591274/cli-command-to-convert-webp-images-to-jpg) who want to do what I want to are displaying lines of code, but I don't really know where to put those lines of code, or how to format them to work with my images in specifc.
If I have the "dwebp" zip folder, I have run the imagemagick installer, and I have a folder of webp files, where do I go from here? What do I need to tell the thing that converts my images, and where can I find that thing that converts my images? I'm feeling very lost. Thanks in advance, and sorry for the noob question.

**Comment by snibgo** [comment]
What version of IM? What is the text output from "magick -version" of "convert -version"? What is your platform (Windows, bash, etc)?

Assuming IM v7, if you have a webp file named in.webp and want to create a png file named out.png:
```
magick in.webp out.png
```
You can use pathnames, eg `\mydir\in.webp` or `/mydir/in.png`

To convert an entire directory, `cd` to that directory, and type this:
```
magick mogrify *.webp
```
Does that help?

**Reply by pigzit** [reply]
Thanks for the quick response! Looks like I'm running v7, [this](https://imgur.com/a/rN28OhM) is what is output when checking, nothing stands out to me but might to you, and I'm running windows 10. Attempting to convert the entire directory went [like this](https://imgur.com/a/l9M92S2). I'm not sure what it means by unable to open.

**Comment by snibgo** [comment, marked as answer]
Sorry, I forgot something. With "`magick mogrify`", you need to specify what output format you want. Do this with "'-fomat PNG'" (or whatever format you want):
```
magick mogrify -format PNG *.webp
```

I don't think your "`cd`" worked, because your prompt stayed at "`C:\Users\Will>`". After doing the "`cd`", try "`dir *.webp`".

**Reply by pigzit** [reply]
> I don't think your "`cd`" worked, because your prompt stayed at "`C:\Users\Will>`". After doing the "`cd`", try "`dir *.webp`".

Oh duh, I was in the wrong drive, didn't even notice that the first time lol.

>Sorry, I forgot something. With "magick mogrify", you need to specify what output format you want. Do this with "'-fomat PNG'" (or whatever format you want):

This worked! Thanks for the help, much appreciated. Maybe I'm a poor navigator, but I struggled to find documentation of each command and their syntax on the IM site itself. Do you have a link or a guide to using the rest of this software?

**Comment by snibgo** [comment]
For `magick` (not `magick mogrify`) see https://imagemagick.org/script/convert.php

For `magick mogrify`, see https://imagemagick.org/script/mogrify.php

For syntax and semantics of options generally, see https://imagemagick.org/script/command-line-options.php

For a more "tutorial" style, see https://www.imagemagick.org/Usage/

Most examples in the official IM documentation use bash. For usage under Windows, see https://imagemagick.org/Usage/windows/ . My own web pages http://im.snibgo.com use mostly Windows BAT, but they don't cover the basics of IM.

**Reply by pigzit** [reply]
Right on, thanks for the help. Much appreciated :-)
